### PR TITLE
inference detached from softmin calculation

### PR DESCRIPTION
**Pull request by mrymsadeghi**
## What?
 
In this PR the softmin score is calculated without inferencing from the model, rather by reading prediction probabilities from a field saved previously to the samples of the dataset
 
## Why?
 
Performing the inference while calculating the metric takes a long time and is not necessayr to be done evertime the metric needs to be calculated
 
## How?
 
The predictions of the panoptic/segmentation model have been previosly saved to the `sample[pred_probs_field]`, the code reads the prediction probabilities instead of running inference on the sample which is computationally more expensive.

 
## Testing
 
Tested on "TRAIN_PANOPTIC_OPTIMIZED_NAT" with 100 samples

image

### CODE REVIEW

File: seametrics/annotations/utils.py [hunk 1]
```diff
@@ -139,68 +153,37 @@ def compute_and_upload_softmin(dataset_view, model_path: str, target_size, mask_
...
-        resized_probs = NNF.interpolate(
-                pred_probs, size=target_size, mode='bilinear', align_corners=False
-            )
-        pred_probs_np = resized_probs.cpu().numpy()
-
-
-        ground_truth_label = create_image_mask(sample['ground_truth_det.detections'], (image.size[1], image.size[0]), label_to_id)
 
-        resized_ground_truth_label = cv2.resize(ground_truth_label, (target_size[1], target_size[0]), interpolation=cv2.INTER_NEAREST)
+        if sample[pred_probs_field] is not None:
```
Comment (franziska-denk) [inline seametrics/annotations/utils.py]: Maybe we could do something like: `if pred_probs_field in sample.field_names and sample[pred_probs_field] is not None:`, because I think that it might raise an error if the pred_probs_field doesnt exist for every sample?

File: seametrics/annotations/utils.py [hunk 2]
```diff
@@ -139,68 +153,37 @@ def compute_and_upload_softmin(dataset_view, model_path: str, target_size, mask_
...
+            pred_probs_np = sample[pred_probs_field] 
+            ground_truth_label = create_image_mask(sample['ground_truth_det.detections'], (target_size[0], target_size[1]), label_to_id)
```
Comment (franziska-denk) [inline seametrics/annotations/utils.py]: Maybe it would be possible to sort the masks inside this function, so that smaller masks are put on top of larger masks, since otherwise boats might be hidden behind water? Or is this already somehow handled and I missed it?

**Review by franziska-denk — COMMENTED** [review]
Hi! I noticed two small details that might be worth adjusting, but other than that it looks good to me!

File: seametrics/annotations/utils.py (hunk 1 above)
Comment (mrymsadeghi) [reply seametrics/annotations/utils.py]: done

File: seametrics/annotations/utils.py (hunk 2 above)
Comment (mrymsadeghi) [reply seametrics/annotations/utils.py]: Good point. I added and tested this change

**Review by franziska-denk — COMMENTED** [review]
looks good to me!

**Review by franziska-denk — APPROVED** [review]
